packages/labs/cli/src/lib/generate/react.ts

```diff
@@ -4,16 +4,202 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {Package} from '@lit-labs/analyzer/lib/model.js';
-
-export const run = async (analysis: Package, console: Console) => {
-  await Promise.all(
-    analysis.modules.map(async (module) => {
-      // For now, log files containing declarations
-      const elements = module.declarations;
-      if (elements.length > 0) {
-        console.log(module.sourceFile.fileName);
+import {
+  ClassDeclaration,
+  LitElementDeclaration,
+  Package,
+  PackageJson,
+} from '@lit-labs/analyzer/lib/model.js';
+import {javascript, FileTree} from './utils.js';
+import * as path from 'path';
+
+const isLitElementDeclaration = (
+  dec: ClassDeclaration
+): dec is LitElementDeclaration => {
+  return (dec as LitElementDeclaration).isLitElement;
+};
+
+interface LitModule {
+  moduleSrcPath: string;
+  moduleJsPath: string;
+  elements: LitElementDeclaration[];
+}
+
+const getLitModules = (packageRoot: string, analysis: Package) => {
+  return analysis.modules
+    .map((module) => {
+      const elements = module.declarations.filter(isLitElementDeclaration);
+      if (elements.length) {
+        const moduleSrcPath = path.relative(
```

Comment:
This logic around figuring out the built (JS) path for a given source file vs. the source path (both relative to the package source) might be a nice thing to move into the analysis itself, since this will likely be needed by all the generators.

Reply:
Went ahead and moved it.